addon/components/frost-textarea.js

```diff
@@ -8,67 +15,71 @@ export default Component.extend(PropTypeMixin, {
   // == Dependencies ==========================================================
 
   // == Properties ============================================================
-
-  attributeBindings: [
-    'autofocus',
-    'cols',
-    'disabled',
-    'placeholder',
-    'readonly',
-    'rows'
-  ],
   classNames: ['frost-textarea'],
   layout,
 
   propTypes: {
     autofocus: PropTypes.bool,
     cols: PropTypes.number,
     disabled: PropTypes.bool,
+    form: PropTypes.string,
     hook: PropTypes.string,
     placeholder: PropTypes.string,
     readonly: PropTypes.bool,
     rows: PropTypes.number,
-    tabindex: PropTypes.number
+    tabindex: PropTypes.number,
+    value: PropTypes.string,
+    wrap: PropTypes.string
   },
 
   getDefaultProps () {
     return {
       autofocus: false,
       disabled: false,
-      tabindex: 0
+      readonly: false,
+      tabindex: 0,
+
+      // Setting these as part of establishing an initial value
+      cols: null,
+      form: null,
+      rows: null,
+      placeholder: null,
+      value: null,
+      wrap: 'soft'
     }
   },
 
   // == Computed Properties ===================================================
 
   @readOnly
-  @computed('disabled', 'value')
+  @computed('disabled', 'readonly', 'value')
   /**
    * Determine whether or not to show button for clearing out text field
    * @param {Boolean} disabled - whether or not input is disabled
+   * @param {Boolean} readonly - whether or not input is readonly
    * @param {String} value - value of text field
    * @returns {Boolean} whether or not to show button for clearing out text field
    */
-  showClear (disabled, value) {
-    return !disabled && Boolean(value)
+  showClear (disabled, readonly, value) {
+    return !disabled && !isEmpty(value) && !readonly
```

Comment:
We put in extra logic in the text component to handle the display/hide of the clear, might be worth duplicating here.  The base concept there was that we only wanted to show the clear when the field had focus, which would remove the need for a `readonly` check (since it can't have focus)